test/common/testlib.py

```diff
@@ -784,9 +785,22 @@ def enable_preload(self, package, *pages):
         self.write_file(path, '{ "preload": [%s]}' % ', '.join('"{0}"'.format(page) for page in pages))
 
     def setUp(self, restrict=True):
+        if os.getenv("MACHINE"):
```

Comment:
this really belongs into `def arg_parse()`, let's please not meddle with `opts` outside of that.

Reply:
I understand I've just tried to do the change as small as possible, so passing via opts was the most straightforward.

Reply:
Yes, it's fine to pass that via `opts`. I just meant, it's better to move that code to the end of `arg_parse()`.

Reply:
arg_parser is not called as part of `setUp()`, but as part of  `test_main`, what is not executed in way of pytest